BBPebbleLib/Pebble.cpp

```diff
@@ -38,46 +43,86 @@ void Pebble::onDisconnect()
 
 void Pebble::onDataReceived(QBluetoothSocket &bt_socket)
 {
-    QDataStream in(&bt_socket);
-
-    if (this->payloadSize == 0) {
-        if (bt_socket.bytesAvailable() < (int)2*sizeof(quint16)) {
+    static const int header_length = (int)2*sizeof(quint16);
+
+    while (bt_socket.bytesAvailable() >= header_length) {
```

Comment:
The loop is definitely unnecessarry because once we have read an entire message, we call again onDataReceived(bt_socket); (Previously line 65, now 86) and could potentially be less efficient within a given thread :)